yt/frontends/gadget/testing.py

```diff
@@ -27,7 +27,7 @@ def write_record(fp, data, endian):
 
 def write_block(fp, data, endian, fmt, block_id):
     assert fmt in [1, 2]
-    block_id = "%-4s" % block_id
```

Comment:
this is the first (and so far only) time I've ever seen this format specifier.
for reference, let me remind reviewers of:
- https://cplusplus.com/reference/cstdio/printf/
- https://fstring.help/